models.js

```diff
@@ -19,4 +19,26 @@ exports.selectArticles = () => {
    ;`
     return db.query(queryStr).then((result)=>{
         return result.rows })
-}
\ No newline at end of file
+}
+
+exports.fetchArticleById = (id) => {
+    //first if checks if NaN
+
+     if(!id.match(/^\d+$/)){
+          return Promise.reject({
+              status:400,
+              msg:'wrong data type for article id'
+          })
+     }
+
+     return db
+     .query(`SELECT * FROM articles 
+      WHERE article_id = $1;
+      `,[id]).then((result)=>{
+        if (result.rows.length === 0){
+            return []
+        } else {
```

Comment:
If the ID that is put in is a number, but the article doesn't exist, then we want to return a 404, not an empty array